rpc-server/src/utils.rs

```diff
@@ -171,10 +175,35 @@ where
     let config = Config::new(CompareMode::Strict).numeric_mode(NumericMode::AssumeFloat);
 
     // Sorts the values of the JSON objects before comparing them.
-    let read_rpc_json = json_sort_value(readrpc_response_json);
+    let read_rpc_json = json_sort_value(read_rpc_response_json);
     let near_rpc_json = json_sort_value(near_rpc_response_json);
 
     if let Err(err) = assert_json_matches_no_panic(&read_rpc_json, &near_rpc_json, config) {
+        // separate mismatching successful and failure responses into different targets
+        // to make it easier to find reasons of the mismatching
+        match response_state {
+            ResponseState::Successful => {
+                tracing::warn!(
+                    target: "shadow_data_consistency_successful_response",
+                    "Shadow data check: DATA MISMATCH\n READ_RPC_DATA: {:?}\n NEAR_RPC_DATA: {:?}\n",
+                    read_rpc_json,
+                    near_rpc_json,
+                );
+            }
+            ResponseState::Failure => {
+                tracing::warn!(target: "shadow_data_consistency_failure_response",
+                    "Shadow data check: DATA MISMATCH\n READ_RPC_STATE: {:?}\n",
```

Comment:
This and the next log message does not explain the situation we have here (who failed, who succeeded)